### PR TITLE
Windows: sync connection status to UI on network changes

### PR DESCRIPTION
**Pull request by alalamav**
* OutlineService sends the connection status to the application on network changes so the UI can be updated accordingly.
  * This was accomplished by keeping a long-lived connection to the service, instead of destroying it after each request. The pipe is only recreated if the service has disconnected.
  * The service keeps the pipe open unless an exception is thrown, in which case it starts listening for requests.
* __TODO__: sign `OutlineService.exe` once the PR is approved.

### CODE REVIEW

File: src/electron/process_manager.ts [hunk 1]
```diff
@@ -76,8 +76,9 @@ const UDP_FORWARDING_TEST_RETRY_INTERVAL_MS = 1000;
 //
 // Fulfills with a copy of `serverConfig` that includes the resolved hostname.
 export function startVpn(
-  serverConfig: cordova.plugins.outline.ServerConfig, onDisconnected: () => void,
-  isAutoConnect = false): Promise<cordova.plugins.outline.ServerConfig> {
+    serverConfig: cordova.plugins.outline.ServerConfig, onDisconnected: () => void,
```
Comment (trevj) [inline src/electron/process_manager.ts]: It seems like there's some redundancy in the two handlers. Did you think about killing `onDisconnected` in favour of listening for reconnection and disconnection events in `onConnectionStatusChange`?

File: src/electron/routing_service.ts [hunk 2]
```diff
@@ -39,18 +39,24 @@ interface RoutingServiceRequest {
...
   errorMessage?: string;
+  connectionStatus: ConnectionStatus;
 }
 
 export interface RoutingService {
-  configureRouting(routerIp: string, proxyIp: string): Promise<void>;
+  configureRouting(
+      routerIp: string, proxyIp: string,
+      onStatusChange: (status: ConnectionStatus) => void): Promise<void>;
+  // configureRouting(routerIp: string, proxyIp: string): Promise<void>;
```
Comment (trevj) [inline src/electron/routing_service.ts]: Guess you need to remove this line :-)

File: tools/OutlineService/OutlineService/OutlineService.cs [hunk 3]
```diff
@@ -738,6 +751,8 @@ private void NetworkAddressChanged(object sender, EventArgs evt)
             {
                 eventLog.WriteEntry("Failed to delete the route to the proxy after a network change.",
                                     EventLogEntryType.Error);
+                ResetRoutingOnNetworkError();
+                return;
```
Comment (trevj) [inline tools/OutlineService/OutlineService/OutlineService.cs]: It looks like any other routes we've added won't be deleted if this fails - that's a change from the current behaviour (and is probably fine) but is it what you intended?

Comment (alalamav) [reply tools/OutlineService/OutlineService/OutlineService.cs]: Which routes are you referring to? `ResetRoutingOnNetworkError` calls `ResetRouting`, which should restore the system routes. Am I missing something?

File: src/electron/routing_service.ts (hunk 2 above)
Comment (alalamav) [reply src/electron/routing_service.ts]: Done in 3ecaa6b.

File: src/electron/process_manager.ts (hunk 1 above)
Comment (alalamav) [reply src/electron/process_manager.ts]: Good suggestion, done in 3ecaa6b.

File: tools/OutlineService/OutlineService/OutlineService.cs (hunk 3 above)
Comment (trevj) [reply tools/OutlineService/OutlineService/OutlineService.cs]: *click* Got it, thanks.

**Pull request merged**